### PR TITLE
[Bugfix] Handle missing `timing_information` gracefully

### PR DESCRIPTION
**Pull request by remibaar**
An error is raised when the devtools before request is not fully executed.
`AttributeError: '_AppCtxGlobals' object has no attribute 'timing_information'`

I've created a test to replicate, and a fix to handle the error gracefully.

More info in #1624 and #1475

## Contributor Checklist

- [x] I have broken down my PR scope into the following TODO tasks
   -  [x] Create test to replicate
   -  [x] Fix bug
- [x] I have run the tests locally and they passed. (refer to testing section in [contributing](https://github.com/plotly/dash/blob/master/CONTRIBUTING.md))
- [x] I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR

### CODE REVIEW

**Comment by remibaar** [comment]
Percy test is failing because of a new string "This is a Copy/Paste friendly version of the traceback." But that's (probably) not due to the changes I've made

File: dash/dash.py [hunk 1]
```diff
@@ -1468,10 +1468,16 @@ def _before_request():
...
-                dash_total = flask.g.timing_information["__dash_server"]
-                dash_total["dur"] = round((time.time() - dash_total["dur"]) * 1000)
+                timing_information = flask.g.get("timing_information", None)
+                if timing_information is None:
+                    return response
 
-                for name, info in flask.g.timing_information.items():
+                timing_information_dash = timing_information.get("__dash_server", None)
+                if timing_information_dash is not None:
+                    dash_total = timing_information["__dash_server"]
```
Comment (alexcjohnson) [inline dash/dash.py:1477]:
```suggestion
                dash_total = timing_information.get("__dash_server", None)
                if dash_total is not None:
```

**Review by alexcjohnson — APPROVED** [review]
Thanks @remibaar - looks good, just a small simplifying suggestion. 💃 

You're right that the Percy change has nothing to do with this PR - strange, we must have inherited a new version of Werkzeug or something? Not a big deal.

**Comment by remibaar** [comment]
Thanks @alexcjohnson  for the simplification! That was exactly how I planned to code it, but for some reason didn't haha. Thanks for catching it.

Do you also agree with the new integration test? I didn't exactly know where to put it, so I've putted in the `test_devtools_ui.py`

**Comment by alexcjohnson** [comment]
The new test is excellent, thanks! I just added a changelog entry, will merge once tests pass.

**Pull request merged**